source/bigint.c

```diff
@@ -958,3 +1048,370 @@ int aws_bigint_shift_left(struct aws_bigint *bigint, size_t shift_amount) {
 
     return AWS_OP_SUCCESS;
 }
+
+/*
+ * Division
+ */
+
+/*
+ * The basic, general algorithm for bigint division requires a divisor to be at least two digits in order for its
+ * quotient digit calculations to be mathematically accurate.  This isn't a big deal since it turns out division by
+ * a single-digit divisor is pretty easy.  That special case is handled here.
+ */
+static int s_aws_bigint_divide_by_single_digit(
+    struct aws_bigint *quotient,
+    struct aws_bigint *remainder,
+    const struct aws_bigint *dividend,
+    uint32_t divisor) {
+
+    AWS_PRECONDITION(aws_bigint_is_valid(quotient));
+    AWS_PRECONDITION(aws_bigint_is_valid(remainder));
+    AWS_PRECONDITION(aws_bigint_is_valid(dividend));
+
+    size_t quotient_length = aws_array_list_length(&dividend->digits);
```

Comment:
Nitpick: const?

EDIT: After reading this a few times, I think const would really help me visualize the moving parts here. I think a quick const pass would be a big help.